File: frontend/src/dto/NoteResponseDto.ts
/**
 * Data Transfer Object (DTO) for representing a Note received from the backend.
 * This structure matches the response body from the backend's note retrieval endpoints.
 */
export interface NoteResponseDto {
  id: number;
  title: string;
  content: string;
  // LocalDateTime in Java maps to string (ISO 8601 format) when sent over HTTP/JSON
  createdAt: string;
  lastModifiedAt: string;
  archived: boolean;
}

File: backend/src/main/java/com/cerdeira/backend/dto/NoteResponseDto.java
package com.cerdeira.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class NoteResponseDto {
    private Long id;
    private String title;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime lastModifiedAt;
    private boolean archived;
}
